cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/MultivariateAnomalyDetection.scala

```diff
@@ -558,81 +501,78 @@ class DetectMultivariateAnomaly(override val uid: String) extends Model[DetectMu
   override def transform(dataset: Dataset[_]): DataFrame = {
     logTransform[DataFrame] {
 
       // check model status first
       try {
-        val responseDict = MADUtils.madGetModel(
-          this.getUrl, this.getModelId, this.getSubscriptionKey).parseJson.asJsObject.fields
+        val response = MADUtils.madGetModel(getUrl, getModelId, getSubscriptionKey)
+          .parseJson.asJsObject.fields
 
-        val modelInfoFields = responseDict("modelInfo").asJsObject.fields
-        val modelStatus = modelInfoFields("status").asInstanceOf[JsString].value.toLowerCase
+        val modelInfo = response("modelInfo").asJsObject.fields
+        val modelStatus = modelInfo("status").asInstanceOf[JsString].value.toLowerCase
         modelStatus match {
-          case "failed" => {
-            val errors = modelInfoFields.get("errors").map(_.convertTo[Seq[DMAError]]).get.toJson.compactPrint
+          case "failed" =>
+            val errors = modelInfo("errors").convertTo[Seq[DMAError]].toJson.compactPrint
             throw new RuntimeException(s"Caught errors during fitting: $errors")
-          }
-          case "created" | "running" => {
-            throw new RuntimeException(s"model ${this.getModelId} is not ready yet")
-          }
-          case "ready" => println("model is ready for inference")
+          case "created" | "running" =>
+            throw new RuntimeException(s"model $getModelId is not ready yet")
+          case "ready" =>
+            logInfo("model is ready for inference")
         }
       } catch {
         case e: RuntimeException =>
-          throw new RuntimeException(s"Encounter error while fetching model ${this.getModelId}, " +
+          throw new RuntimeException(s"Encounter error while fetching model $getModelId, " +
             s"please double check the modelId is correct: ${e.getMessage}")
       }
 
-      val blobContainerClient = getBlobContainerClient
-      val df = dataset.select((Array(getTimestampCol) ++ getInputCols).map(col): _*)
-        .sort(col(getTimestampCol).asc).toDF()
-      val sasUrl = upload(blobContainerClient, df)
-
-      val httpRequestBase = prepareRequest(prepareEntity(sasUrl).get)
-      val request = new HTTPRequestData(httpRequestBase.get)
-      val response = handlingFunc(Client, request)
+      val spark = dataset.sparkSession
+      val responseJson = submitDataset(dataset)
 
-      val responseJson = IOUtils.toString(response.entity.get.content, "UTF-8")
-        .parseJson.asJsObject.fields
+      val summary = responseJson("summary").convertTo[DMASummary]
 
-      val summary = responseJson.get("summary").map(_.convertTo[DMASummary]).get
-      if (summary.status == "FAILED") {
+      if (summary.status.toLowerCase() == "failed") {
         val errors = summary.errors.get.toJson.compactPrint
-        throw new RuntimeException(s"Caught errors during inference: $errors")
+        throw new RuntimeException(s"Could not get trained model: $errors")
```

Comment:
This is not `can't get trained model` error I guess, it should be inference failure?

Reply:
Fixed